client/src/components/ElementEditor/ElementActions.js

```diff
@@ -74,7 +93,16 @@ class ElementActions extends Component {
    * @returns {ActionMenuComponent|null}
    */
   render() {
-    const { children, id, ActionMenuComponent } = this.props;
+    const { children, id, ActionMenuComponent, inlineEditable } = this.props;
+
+    // Flatten the recursive structure of children - [[[..., react.element], react.element]
+    // and filter AbstractActions's that can be shown depending if the Element is inlineEditable
+    let kids = children;
```

Comment:
Can we find a more descriptive variable name than `kids`, representing why it's different from `children`?

Reply:
I think I just used kids as a placeholder variable since I couldn't think of anything better at the time and intended to change it later.  Guess I forgot to :)

Reply:
I really don't like this approach. This seems like an anti-pattern.

Element actions shouldn't have to worry about filtering out it's children. The `showForNonInlineEditableBlock` prop is literally not used in the `AbstractAction` ... it's just there as a flag to be filtered out here.